dScripts/ai/MINIGAME/SG_GF/SERVER/SGCannon.cpp

```diff
@@ -673,7 +673,7 @@ void SGCannon::RegisterHit(Entity* self, Entity* target, const std::string& time
 		target->GetPosition()
 	);
 
-	auto newScore = (int)self->GetVar<uint32_t>(TotalScoreVariable) + score;
+	auto newScore = self->GetVar<int32_t>(TotalScoreVariable) + score;
```

Comment:
This actually must stay as uint32_t.  Variables on scripts are attached by not just key, but also T type.

Reply:
I tried doing that, but it enables a roll-under bug if you shoot a friendly ship.

Reply:
Change all of the total score to be stored as int in that case.